library/data/registry_entry_attributes/src/main/java/org/quiltmc/qsl/registry/attribute/impl/RegistryEntryAttributeImpl.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.registry.attribute.impl;
+
+import com.mojang.serialization.Codec;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+import org.quiltmc.qsl.registry.attribute.api.RegistryEntryAttribute;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+
+@ApiStatus.Internal
+public record RegistryEntryAttributeImpl<R, V>(Registry<R> registry,
+											   Identifier id,
+											   Side side,
+											   Codec<V> codec,
+											   @Nullable V defaultValue,
+											   @Nullable Function<R, V> missingValueFunction)
+		implements RegistryEntryAttribute<R, V> {
+	@Override
+	public Optional<V> getValue(R entry) {
```

Comment:
Please have empty lines before and after a new scope, here everything being so close together makes it hard to read.